src/main/scala/mesosphere/marathon/core/matcher/manager/impl/OfferMatcherManagerActor.scala

```diff
@@ -174,7 +174,7 @@ private[impl] class OfferMatcherManagerActor private (
       .map(_.getDisk.getPersistence.getId)
       .collect { case LocalVolumeId(volumeId) => volumeId.runSpecId }
       .toSet
-    val (reserved, normal) = matchers.toSeq.partition(_.precedenceFor.exists(appReservations))
+    val (reserved, normal) = matchers.toSeq.filter(_.isInterestedIn(offer)).partition(_.precedenceFor.exists(appReservations))
```

Comment:
As I remember the proposal was to register with an interest in offers or not. What is the motivation to ask the offer matcher instead of being told by it? I like this approach more but am just curious.

Reply:
In the end I moved toward this approach as I found it fitting what we wanted to do the most... But I am up for a discussion if you or someone else would approach it differently...

Reply:
Hm, @meichstedt proposed the filter on registration approach. My vote would go to yours.